plugins/modules/vmware_drs_override.py

```diff
@@ -0,0 +1,156 @@
+
+DOCUMENTATION = '''
+---
+module: vmware_drs_override
+short_description: Configure DRS behavior for a specific VM in vSphere
+description:
+    - This module allows setting a DRS behavior override for individual VMs within a DRS-enabled VMware vSphere cluster.
+options:
+    hostname:
+        description:
+            - The hostname or IP address of the vCenter server.
+        required: true
+        type: str
+    username:
+        description:
+            - The username for vCenter authentication.
+        required: true
+        type: str
+    password:
+        description:
+            - The password for vCenter authentication.
+        required: true
+        type: str
+    port:
+        description:
+            - The port number for the vCenter server.
+        required: false
+        type: int
+        default: 443
+    validate_certs:
+        description:
+            - If C(false), SSL certificates will not be validated.
+        type: bool
+        default: False
+    vm_name:
+        description:
+            - Name of the VM for which the DRS override is set.
+        required: true
+        type: str
+    drs_behavior:
+        description:
+            - Desired DRS behavior for the VM.
+        choices: ['manual', 'partiallyAutomated', 'fullyAutomated']
+        default: 'manual'
+        type: str
+author:
+    - Your Name (@your_github_username)
+'''
+
+EXAMPLES = '''
+- name: Set DRS behavior for a VM
+  vmware_drs_override:
+    hostname: "vcenter.example.com"
+    username: "administrator@vsphere.local"
+    password: "yourpassword"
+    port: 443
+    validate_certs: False
+    vm_name: "my_vm_name"
+    drs_behavior: "manual"
+'''
+
+RETURN = '''
+changed:
+    description: Whether the DRS behavior was changed.
+    type: bool
+    returned: always
+msg:
+    description: A message describing the outcome of the task.
+    type: str
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.vmware.plugins.module_utils.vmware import (
+    PyVmomi,
+    connect_to_api,
+    vmware_argument_spec,
+)
+from pyVmomi import vim, vmodl
+
+
+class VmwareDrsOverride(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDrsOverride, self).__init__(module)
+        self.module = module
+        self.vm_name = module.params['vm_name']
+        self.drs_behavior = module.params['drs_behavior']
+        self.content = self.si.RetrieveContent()
+
+    def get_vm_by_name(self):
+        obj_view = self.content.viewManager.CreateContainerView(self.content.rootFolder, [vim.VirtualMachine], True)
+        for vm in obj_view.view:
+            if vm.name == self.vm_name:
+                obj_view.Destroy()
+                return vm
+        obj_view.Destroy()
+        self.module.fail_json(msg="VM '%s' not found." % self.vm_name)
+
+    def set_drs_override(self):
+        vm = self.get_vm_by_name()
+        cluster = vm.runtime.host.parent
```

Comment:
Need check for cluster without DRS, for standalone esxi in vCenter, single standalone esxi. The PyVmomi module has special functions for this (is_vcenter, for example).

And for this module need some integration tests.

Reply:
Will take a look as well. Thanks!